external/avro/src/main/scala/org/apache/spark/sql/avro/AvroOutputWriterFactory.scala

```diff
@@ -40,6 +40,8 @@ private[sql] class AvroOutputWriterFactory(
       path: String,
       dataSchema: StructType,
       context: TaskAttemptContext): OutputWriter = {
+    context.getConfiguration.set("avro.serialization.data.model",
+      "org.apache.avro.generic.GenericData")
```

Comment:
This is a simple change to use GenericDatumReader in the avro lib instead of the ReflectDatumReader (which is the default). Write time improvements are in the attached pdf, pg 3 under column A.